pal/samples/esp8266/telemetry_sample.c.ino

```diff
@@ -0,0 +1,232 @@
+            IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_TRUSTED_CERT, certificates);
+#endif // SET_TRUSTED_CERT_IN_SAMPLES
+
+#if defined SAMPLE_MQTT || defined SAMPLE_MQTT_WS
+        //Setting the auto URL Encoder (recommended for MQTT). Please use this option unless
+        //you are URL Encoding inputs yourself.
+        //ONLY valid for use with MQTT
+        bool urlEncodeOn = true;
+        IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_AUTO_URL_ENCODE_DECODE, &urlEncodeOn);
+        /* Setting Message call back, so we can receive Commands. */
+        if (IoTHubClient_LL_SetMessageCallback(device_ll_handle, receive_message_callback, &receiveContext) != IOTHUB_CLIENT_OK)
+        {
+            LogInfo("ERROR: IoTHubClient_LL_SetMessageCallback..........FAILED!\r\n");
+        }
+#endif
+
+        // Setting connection status callback to get indication of connection to iothub
+        (void)IoTHubDeviceClient_LL_SetConnectionStatusCallback(device_ll_handle, connection_status_callback, NULL);
+
+        do
+        {
+            if (messages_sent < MESSAGE_COUNT)
+            {
+                // Construct the iothub message from a string or a byte array
+                message_handle = IoTHubMessage_CreateFromString(telemetry_msg);
+                //message_handle = IoTHubMessage_CreateFromByteArray((const unsigned char*)msgText, strlen(msgText)));
+
+                // Set Message property
+                /*(void)IoTHubMessage_SetMessageId(message_handle, "MSG_ID");
+                (void)IoTHubMessage_SetCorrelationId(message_handle, "CORE_ID");
+                (void)IoTHubMessage_SetContentTypeSystemProperty(message_handle, "application%2fjson");
+                (void)IoTHubMessage_SetContentEncodingSystemProperty(message_handle, "utf-8");*/
+
+                // Add custom properties to message
+//                (void)IoTHubMessage_SetProperty(message_handle, "property_key", "property_value");
+
+                LogInfo("Sending message %d to IoTHub\r\n", (int)(messages_sent + 1));
+                result = IoTHubDeviceClient_LL_SendEventAsync(device_ll_handle, message_handle, send_confirm_callback, NULL);
+                // The message is copied to the sdk so the we can destroy it
+                IoTHubMessage_Destroy(message_handle);
+
+                messages_sent++;
+            }
+            else if (g_message_count_send_confirmations >= MESSAGE_COUNT)
+            {
+                // After all messages are all received stop running
+                g_continueRunning = false;
+            }
+
+            IoTHubDeviceClient_LL_DoWork(device_ll_handle);
+            ThreadAPI_Sleep(3);
+
+        } while (g_continueRunning);
+
+        // Clean up the iothub sdk handle
+        IoTHubDeviceClient_LL_Destroy(device_ll_handle);
+    }
+    // Free all the sdk subsystem
+    IoTHub_Deinit();
+
+    LogInfo("done with sending");
+    return;
+}
+
+void loop(void)
+{
+  
+}
```

Comment:
any need for this?

Reply:
I think you may need to have a loop function for Arduino even if you don't use it.